libass/ass_font.c

```diff
@@ -457,8 +458,16 @@ static int add_face(ASS_FontSelector *fontsel, ASS_Font *font, uint32_t ch)
     set_font_metrics(face);
 
     font->faces[font->n_faces] = face;
-    font->faces_uid[font->n_faces++] = uid;
-    return font->n_faces - 1;
+    font->faces_uid[font->n_faces] = uid;
+    if (!ass_create_hb_font(font, font->n_faces)) {
+        FT_Done_Face(face);
+        goto fail;
```

Comment:
That looks unnecessary complicated. Something like
```c
    if (ass_create_hb_font(font, font->n_faces))
        return font->n_faces++;
    FT_Done_Face(face);
    return -1;
```
should be better.

Reply:
The fail path will later be needed for threading (to unlock), and I don't see a particular reason to reorganize this here only to redo it again in the threading branch.